### PR TITLE
Add tests for 100% coverage of Repository.Person.equals() - Fix #17

### LINKED ISSUE #17: Tests for Repository.Person.equals()

### PR DESCRIPTION
**Pull request by vronc**
Add tests to RepositoryTest to improve coverage on method Person.equals.

Previously tested requirements of Person.equals() (Coverage 24%) :
- Comparision of age.
- Positive instance, when compared objects are equal.
 Previously untested but now tested requirements of Person.equals() (Coverage 100%):
- Positive instances:
    - Compare object to self.
    - Compare equal objects that don't have first names or surnames.
- Negative instances:
    - Compare non-equal first names, surnames and id's.
    - Compare if one Person-object's first name, surname or id is null.
    - If input object is null.
    - If the input object is not of class Person.

### CODE REVIEW

File: repository/src/test/java/com/iluwatar/repository/RepositoryTest.java [hunk 1]
```diff
@@ -119,9 +130,80 @@ public void testFindOneByNameEqualSpec() {
...
+    p2.setName(null);
+    assertTrue(p1.equals(p2));
+
+    p1.setName("John");
+    p2.setName("John");
+    //Two Person-objects with no surnames
+    p1.setSurname(null);
+    p2.setSurname(null);
+    assertTrue(p1.equals(p2));
+  }
```
Comment (alzahraasalman) [inline repository/src/test/java/com/iluwatar/repository/RepositoryTest.java:158]: This test should be divided to 3 different tests with one asserTrue each

**Review by alzahraasalman — CHANGES_REQUESTED** [review]

File: repository/src/test/java/com/iluwatar/repository/RepositoryTest.java [hunk 2]
```diff
@@ -119,9 +130,80 @@ public void testFindOneByNameEqualSpec() {
...
+
+    //null-comparision
+    assertFalse(p1.equals(null));
+
+    //Comparision with object from other class
+    int notAPerson = 1;
+    assertFalse(p2.equals(notAPerson));
+
+  }
+
```
Comment (alzahraasalman) [inline repository/src/test/java/com/iluwatar/repository/RepositoryTest.java:224]: Same here, split into several tests :)

**Review by alzahraasalman — CHANGES_REQUESTED** [review]

**Review by alzahraasalman — APPROVED** [review]
Looks good!

**Comment by vronc** [comment]
Fix #17

**Pull request merged**